lib/pbench/client/__init__.py

```diff
@@ -346,8 +346,8 @@ def create_api_key(self):
         Creating an API key will cause the new key to be used instead of a
         normal login auth_token until the API key is removed.
         """
-        response = self.post(api=API.KEY)
-        self.api_key = response.json()["api_key"]
+        response = self.post(api=API.KEY, uri_params={"key": ""})
```

Comment:
This is an artifact of the client library templatization code. Basically there's no practical way to indicate whether a URI parameter is "optional" and it (in this case over cautiously) fails if no parameters are provided.

Some alternatives I decided not to pursue:
1. The client could *know* which are optional, which ties the client more closely to the implementation... and I don't really like that.
2. We could figure out a way for `endpoints` to determine through Flask which parameters are optional, and encode that into the `uri` directory. (E.g., we already specify the Flask "type" of each, and we could add `optional=[True|False]` if we worked out a way to tell.)
3. The client could silently remove all template parameters if no values are provided, then try the call and see whether it fails. It has the necessary information since the `endpoints` directory names all of the parameters. I didn't find this entirely satisfying, but it'd work.
4. Require the caller to *acknowledge* that there are parameters it's choosing not to set, using the `{"key": ""}` format seen here. I admit I chose this more because it didn't require any framework changes than for any other reason, but I don't think it's unreasonable.

Reply:
Apparently I'm late to this party -- [Dave answered this last week](https://github.com/distributed-system-analysis/pbench/pull/3410#issuecomment-1526389360):  it's a "feature" of the Pbench client code, not a design failure in the Server API....  👃 🤏